blender_addons/io_md5_exporter/io_export_md5.py

```diff
@@ -993,86 +1013,256 @@ def invoke(self, context, event):
+      data["Stand"]["animationPool"].append(modelName+a.name)
+
+  # assetDirectory = getTargetTerasologyAssetsDirectory(bpy.context)
+  prefabDirectory = os.path.join(assetDirectory,"prefabs")
   fileName = modelName+".prefab"
+  print(fileName)
   createDirectoryForFileIfMissing(os.path.join(prefabDirectory,fileName))
-  print("Exporting prefab at: "+prefabDirectory)
+  # print("Exporting prefab at: " + prefabDirectory)
   with open(os.path.join(prefabDirectory,fileName),"w") as outFile:
     json.dump(data, outFile, indent=2)
 
   return None
 
+def exportMaterialForTerasology(assetDirectory, report = print):
+  current_file_dir = os.path.dirname(__file__)
+  file_in_path=os.path.join(current_file_dir,"templateSkin.mat")
+  with open(file_in_path, 'r') as inFile:
+    data = json.load(inFile, object_pairs_hook = OrderedDict)
+
+  modelName = getSuggestedModelName()
+
+  data["params"]["diffuse"] = modelName.lower()
+
+  # assetDirectory = getTargetTerasologyAssetsDirectory(bpy.context)
+  materialDirectory = os.path.join(assetDirectory,"materials")
+  fileName = modelName.lower()+"Skin.mat"
+  createDirectoryForFileIfMissing(os.path.join(materialDirectory,fileName))
+  print("Exporting material file at: " + materialDirectory)
+  with open(os.path.join(materialDirectory,fileName),"w") as outFile:
+    json.dump(data, outFile, indent = 2)
+
+  return None
+
+# import getpass
+def exportModuleForTerasology(parentDirectory):
+  data = json.loads(moduleData, object_pairs_hook = OrderedDict)
+  print(data)
+  modelName  = getSuggestedModelName()
+  data["id"] = modelName[:1].upper()+modelName[1:]
+  data["displayName"] = modelName
+  data["description"] = "This module contains a single " + modelName + "model that can be spawned with spawnPrefab" + modelName + ". The module has been generated with the Blender addon from the TeraMisc repository"
+  # data["author"] = getpass.getuser()
+  data["version"] = "0.1.1-SNAPSHOT"
+
+  fileName = "modules.txt"
+  with open(os.path.join(parentDirectory, fileName),'w') as outFile:
+    json.dump(data, outFile, indent = 2)
+
+  return None
+
+
+
+from shutil import copy
+def exportTextureForTerasology(assetDirectory, report = print):
+  blendFilePath = bpy.path.abspath("//")
+  modelName = getSuggestedModelName()
+  textureFileName = modelName[:1].upper()+modelName[1:] + "Texture.png"
+  texturePath = os.path.join(blendFilePath,textureFileName)
+  texturePath2 = bpy.path.abspath(bpy.data.images[0].filepath)
+  print("++++++++++++",bpy.path.abspath(bpy.data.images[0].filepath))
+  if not os.path.isfile(texturePath) or not os.path.isfile(texturePath2):
+    report({'ERROR'}, "Texture File must be present in same folder as the .blend file")
+    return
+
+  # assetDirectory = getTargetTerasologyAssetsDirectory(bpy.context)
+  textureDirectory = os.path.join(assetDirectory,"textures")
+  if not os.path.exists(textureDirectory):
+    os.makedirs(textureDirectory)
+
+  if os.path.isfile(texturePath2):

```

Comment:
I think previously the actual texture path should be determined so that this complexity here is not necessary.

I think it would be a good idea to actually extract the logic that determines the texture path. 
e.g. you could introduce a method called getTexturePath the the documentation "returns the location of the texture or when none can be found the location where the texture should be".